.github/workflows/ci.yml

```
@@ -2,6 +2,9 @@ name: CI pipeline
 
 on: [ push, pull_request, workflow_dispatch ]
 
+permissions:
+  contents: read
```

```suggestion
permissions: read-all
```
According to the docs you linked, this seems more appropriate as these actions shouldn't change anything outside of showing an output